### DISCUSSION TITLE
Creating 3D Cylinder Mesh for Porous Flow Simulation

### DISCUSSION THREAD

**Discussion by ColinMG2** [idaholab/moose#29130] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello. I am a new user to MOOSE and I am a little confused on how to test whether the functions I am using are creating a cylindrical mesh. I want to simulate Fick's second law of diffusion for a 1D case where diffusion only occurs in the z-direction: 
![image](https://github.com/user-attachments/assets/e2719ca5-ea8f-4b21-9ac1-6697055d9aa1)
My current code is as follows:
```
[Mesh]
    [cross_section]
        type = ConcentricCircleMeshGenerator
        has_outer_square = off
        num_sectors = 4
        preserve_volumes = off
        radii = 1
        rings = 1
    []
    [Extrusion]
        type = AdvancedExtruderGenerator
        direction = '0 0 1'
        heights = '1 2 3'
        input = cross_section
        num_layers = '1 2 3'
    []
[]
```
I am not quite sure how to test this as when I tried to run just the mesh I got this error: 
(moose) colinmoose@ColinG14:~/projects/moosecmg$ ./moosecmg-opt -i porous_flow/step1.i --mesh-only

*** ERROR ***
The following error occurred in the Action '' of type SetupMeshCompleteAction.

No mesh file was supplied and no generation block was provided

How would you recommend starting with the mesh for this problem?

**Comment by GiudGiud** [comment]
Hello

This syntax should be fine. Maybe something else in the input is really breaking the parser
Can you please pase the rest of the input?

**Reply by ColinMG2** [reply]
This is what I ran in the command terminal and the corresponding output. I am trying to test the mesh only:
(moose) colinmoose@ColinG14:~/projects/moosecmg$ ./moosecmg-opt -i porous_flow/step1.i --mesh-only


*** ERROR ***
The following error occurred in the Action '' of type SetupMeshCompleteAction.

No mesh file was supplied and no generation block was provided

Stack frames: 14
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: MooseBase::callMooseError(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) const
3: /home/colinmoose/projects/moose/framework/libmoose-opt.so.0(+0xbe109d) [0x7fb27176809d]
4: SetupMeshCompleteAction::act()
5: Action::timedAct()
6: ActionWarehouse::executeActionsWithAction(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
7: ActionWarehouse::executeAllActions()
8: MooseApp::runInputFile()
9: MooseApp::run()
10: ./moosecmg-opt(+0x20e8) [0x5604234510e8]
11: /lib/x86_64-linux-gnu/libc.so.6(+0x2a1ca) [0x7fb2632ab1ca]
12: __libc_start_main
13: ./moosecmg-opt(+0x238a) [0x56042345138a]
Abort(1) on node 0 (rank 0 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0

The moose file input is the same as what I provided in the original post.

**Reply by GiudGiud** [reply]
This input with just the two generators runs fine on my machine
```
(moose)$ ~/projects/moose_v11/test/moose_test-devel -i mesh.i --mesh-only
 Mesh Information:
  elem_dimensions()={3}
  spatial_dimension()=3
  n_nodes()=287
    n_local_nodes()=287
  n_elem()=192
    n_local_elem()=192
    n_active_elem()=192
  n_subdomains()=1
  n_elemsets()=0
  n_partitions()=1
  n_processors()=1
  n_threads()=1
  processor_id()=0
  is_prepared()=true
  is_replicated()=true

 Mesh Bounding Box:
  Minimum: (x,y,z)=(      -1,       -1,        0)
  Maximum: (x,y,z)=(       1,        1,        6)
  Delta:   (x,y,z)=(       2,        2,        6)

 Mesh Element Type(s):
  HEX8

 Mesh Nodesets:
  Nodeset 0, 41 nodes
   Bounding box minimum: (x,y,z)=(      -1,       -1,        0)
   Bounding box maximum: (x,y,z)=(       1,        1,        0)
   Bounding box delta: (x,y,z)=(       2,        2,        0)
  Nodeset 1, 41 nodes
   Bounding box minimum: (x,y,z)=(      -1,       -1,        6)
   Bounding box maximum: (x,y,z)=(       1,        1,        6)
   Bounding box delta: (x,y,z)=(       2,        2,        0)

 Mesh Sidesets:
  Sideset 0, 32 sides (QUAD4), 32 elems (HEX8), 41 nodes
   Side volume: 3.06147
   Bounding box minimum: (x,y,z)=(      -1,       -1,        0)
   Bounding box maximum: (x,y,z)=(       1,        1,        0)
   Bounding box delta: (x,y,z)=(       2,        2,        0)
  Sideset 1, 32 sides (QUAD4), 32 elems (HEX8), 41 nodes
   Side volume: 3.06147
   Bounding box minimum: (x,y,z)=(      -1,       -1,        6)
   Bounding box maximum: (x,y,z)=(       1,        1,        6)
   Bounding box delta: (x,y,z)=(       2,        2,        0)

 Mesh Edgesets:
  None

 Mesh Subdomains:
  Subdomain 1: 192 elems (HEX8, 192 active), 287 active nodes
   Volume: 18.3688
   Bounding box minimum: (x,y,z)=(      -1,       -1,        0)
   Bounding box maximum: (x,y,z)=(       1,        1,        6)
   Bounding box delta: (x,y,z)=(       2,        2,        6)
  Global mesh volume = 18.3688

(moose)$ ll
total 48
drwxr-xr-x   4 giudgl  898089231    128 Nov 25 06:52 ./
drwxr-xr-x  77 giudgl  898089231   2464 Nov 25 06:52 ../
-rw-r--r--   1 giudgl  898089231    381 Nov 25 06:52 mesh.i
-rw-r--r--   1 giudgl  898089231  19672 Nov 25 06:52 mesh_in.e
(moose) giudgl@FN654105:~/projects/Studies/Discussions/ColinMG$ cat mesh.i 
[Mesh]
    [cross_section]
        type = ConcentricCircleMeshGenerator
        has_outer_square = off
        num_sectors = 4
        preserve_volumes = off
        radii = 1
        rings = 1
    []
    [Extrusion]
        type = AdvancedExtruderGenerator
        direction = '0 0 1'
        heights = '1 2 3'
        input = cross_section
        num_layers = '1 2 3'
    []
[]
```

**Reply by GiudGiud** [reply]
what is the commit you are using for moose?
in moose/ `git show` will show that

**Reply by ColinMG2** [reply]
This is what it says when I run it in my app name:
(moose) colinmoose@ColinG14:~/projects/moosecmg$ git show
fatal: your current branch 'master' does not have any commits yet

**Reply by GiudGiud** [reply]
let's try in your moose directory not in your app

**Reply by ColinMG2** [reply]
It worked in my moose directory. Got the same output as you:
(moose) colinmoose@ColinG14:~/projects/moose/test$ ./moose_test-opt -i test.i --mesh-only
 Mesh Information:
  elem_dimensions()={3}
  spatial_dimension()=3
  n_nodes()=287
    n_local_nodes()=287
  n_elem()=192
    n_local_elem()=192
    n_active_elem()=192
  n_subdomains()=1
  n_elemsets()=0
  n_partitions()=1
  n_processors()=1
  n_threads()=1
  processor_id()=0
  is_prepared()=true
  is_replicated()=true

 Mesh Bounding Box:
  Minimum: (x,y,z)=(      -1,       -1,        0)
  Maximum: (x,y,z)=(       1,        1,        6)
  Delta:   (x,y,z)=(       2,        2,        6)

 Mesh Element Type(s):
  HEX8

 Mesh Nodesets:
  Nodeset 0, 41 nodes
   Bounding box minimum: (x,y,z)=(      -1,       -1,        0)
   Bounding box maximum: (x,y,z)=(       1,        1,        0)
   Bounding box delta: (x,y,z)=(       2,        2,        0)
  Nodeset 1, 41 nodes
   Bounding box minimum: (x,y,z)=(      -1,       -1,        6)
   Bounding box maximum: (x,y,z)=(       1,        1,        6)
   Bounding box delta: (x,y,z)=(       2,        2,        0)

 Mesh Sidesets:
  Sideset 0, 32 sides (QUAD4), 32 elems (HEX8), 41 nodes
   Side volume: 3.06147
   Bounding box minimum: (x,y,z)=(      -1,       -1,        0)
   Bounding box maximum: (x,y,z)=(       1,        1,        0)
   Bounding box delta: (x,y,z)=(       2,        2,        0)
  Sideset 1, 32 sides (QUAD4), 32 elems (HEX8), 41 nodes
   Side volume: 3.06147
   Bounding box minimum: (x,y,z)=(      -1,       -1,        6)
   Bounding box maximum: (x,y,z)=(       1,        1,        6)
   Bounding box delta: (x,y,z)=(       2,        2,        0)

 Mesh Edgesets:
  None

 Mesh Subdomains:
  Subdomain 1: 192 elems (HEX8, 192 active), 287 active nodes
   Volume: 18.3688
   Bounding box minimum: (x,y,z)=(      -1,       -1,        0)
   Bounding box maximum: (x,y,z)=(       1,        1,        6)
   Bounding box delta: (x,y,z)=(       2,        2,        6)
  Global mesh volume = 18.3688

How do I properly set up git for my app so that it will work in that directory?

**Reply by GiudGiud** [reply]
If you want git show to show something in your folder you need to make a commit first. 

For fixing your app I would recommend recompiling, possibly after cleaning the repository

**Reply by ColinMG2** [reply]
Okay perfect thank you! That worked! I am now having trouble understanding how to properly apply boundary conditions for the problem since I don't know how to name them correctly. Here is my moose file now:
[Mesh]
    [cross_section]
       type = ConcentricCircleMeshGenerator
       has_outer_square = off
       num_sectors = 6
       preserve_volumes = on
       radii = 1
       rings = 1
    []
    [Extrusion]
        type = AdvancedExtruderGenerator
        direction = '0 0 1'
        heights = '0.5 0.5 0.5 0.5 0.5 0.5'
        input = cross_section
        num_layers = '1 1 1 1 1 1'
    []
[]

[Variables]
    [concentration]
    []
[]

[Kernels]
    [diffusion]
        type = ADDiffusion
        variable = concentration
    []
[]

[BCs]
    [bottom]
        type = DirichletBC
        boundary = bottom
        value = 536
        variable = concentration
    []
    [sides]
        type = NeumannBC
        boundary = sides
        variable = concentration
    []
    [top]
        type = NeumannBC
        boundary = top
        variable = concentration
    []
[]

[Executioner]
  type = Transient
[]

[Outputs]
  exodus = true
[]

It says that the following node sets (ids) do not exist on the mesh: bottom
